clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2018,6 +2018,11 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
       return true;
     break;
   }
+  case Builtin::BI__builtin_hlsl_group_memory_barrier_with_group_sync: {
+    if (SemaRef.checkArgCountAtMost(TheCall, 0))
```

Comment:
Is there a reason to not use `SemaRef.checkArgCount(TheCall, 0)`? I think that would be more descriptive.

Reply:
The reason is `checkArgCount` actually expects there to be at least one arg. I opted to just call a different function rather than modifying `checkArgCount`.

Reply:
Do we even need to check the argument count? since the prototype is `void()` instead of being variadic we shouldn't even be able to get here if someone tries to call the builtin with an argument - I would expect that they'd get an error about having too many arguments way before this.

Reply:
Verified that removing this check the error still works.